packages/compass-e2e-tests/helpers/commands/scroll-to-virtual-item.ts

```diff
@@ -0,0 +1,53 @@
+import type { CompassBrowser } from '../compass-browser';
+
+export async function scrollToVirtualItem(
+  browser: CompassBrowser,
+  itemsSelector: string,
+  targetSelector: string
+): Promise<boolean> {
+  let found = false;
+
+  let lastId: string;
+
+  // Start by scrolling down, then scroll back up. This is necessary in case we
+  // started off somewhere in the middle and the item we're looking for is above
+  // it.
+  for (const direction of [1, -1]) {
+    await browser.waitUntil(async () => {
+      const targetElement = await browser.$(targetSelector);
+      if (await targetElement.isExisting()) {
+        await targetElement.waitForDisplayed();
+        await targetElement.scrollIntoView();
+        // the item is now visible, so stop scrolling
+        found = true;
+        return true;
+      }
+
+      const elements = await browser.$$(itemsSelector);
+      const edgeElement = elements[direction === 1 ? elements.length - 1 : 0];
+      await edgeElement.waitForDisplayed();
+      await edgeElement.scrollIntoView();
+
+      const thisId = await edgeElement.getAttribute('data-id');
+
+      if (thisId === lastId) {
```

Comment:
probably not important and i'm not sure this can happen, but what would happen if both are undefined (like someone removed `data-id`)

Reply:
waitUntil will time out after 5 seconds, twice.

Reply:
If we looked up the card as `[data-testid="collection-grid-item"][data-id="${dbName}.${collectionName}"]` and waited for it to exist or be displayed it would time out the exact same way if it never existed, so I think that's fine :)